client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClient.java

```diff
@@ -166,14 +162,14 @@ protected static final void setApplicationProperties(final Message msg, final Ma
             sender.openHandler(senderOpen -> {
                 if (senderOpen.succeeded()) {
                     LOG.info("sender open [{}] sendQueueFull [{}]", sender.getRemoteTarget(), sender.sendQueueFull());
-                    // wait on credits a little time, if not already given TODO: configurable waiting time
+                    // wait on credits a little time, if not already given
                     if (sender.sendQueueFull()) {
-                        ctx.owner().setTimer(10, timerID -> {
-                            LOG.info("waited 10ms on credits [{}]", sender.getCredit());
-                            result.complete(senderOpen.result());
+                        ctx.owner().setTimer(waitForInitialCredits, timerID -> {
```

Comment:
sanity check? `waitForInitialCredits` >= 0? 0 = wait forever?

Reply:
The timer from vertx already does a check, that it need to be >= 1.

Reply:
and what happens if it isn't? is an exception thrown? if so, then it should be declared in the JavaDocs ...